File: newton.cpp
#include <iostream>
#include <ctime>
#include <cmath>

using namespace std;

double g(double x)
{
	double gx = .5 * x * x + x + 1 - exp(x);
	return gx;
}

double gp(double x)
{
	double gpx = x + 1 - x * exp(x);
	return gpx;
}

double newtong(double xn)
{
	double ans = xn - (g(xn)) / (gp(xn));
	return ans;
}

int main()
{
	double x = 100;
	double ans = 0;
	int itr = 0;
	double tolerance = 0.001;

	time_t start = time(nullptr);
	while(abs(x - ans) > tolerance)
	{
		x = newtong(x);
		itr++;	
	}
	time_t end = time(nullptr);	

	cout << "The root was found to be approximately " << x << " in " << itr << " iterations and " << end - start << " seconds." << endl;

	return 0;
}

File: newton.py
import time
import math
import decimal

def g(x):
    gx = decimal.Decimal(.5) * decimal.Decimal(x) * decimal.Decimal(x) + decimal.Decimal(x) + decimal.Decimal(1) - decimal.Decimal(math.e) ** decimal.Decimal(x)
    return gx

def gp(x):
    gpx = decimal.Decimal(x) + decimal.Decimal(1) - decimal.Decimal(x) * decimal.Decimal(math.e) ** decimal.Decimal(x)
    return gpx

def newtong(xn):
    return decimal.Decimal(xn) - (decimal.Decimal(g(decimal.Decimal(xn)))) / (decimal.Decimal(gp(decimal.Decimal(xn))))

x = decimal.Decimal(100.0)
ans = decimal.Decimal(0)
itr = 0
tolerance = decimal.Decimal(0.001)

start = time.time()
while(abs(decimal.Decimal(x) - decimal.Decimal(ans)) > decimal.Decimal(tolerance)):
    x = decimal.Decimal(newtong(decimal.Decimal(x)))
    itr = itr + 1
end = time.time()

print("The root was found to be approximately " + str(decimal.Decimal(x)) + " in " + str(itr) + " iterations and " + str(end - start) + " seconds.")
